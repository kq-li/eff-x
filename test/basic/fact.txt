fact : int -> int = \x : int . {
  acc : int = 1;
  while (gt x 1) {
    acc : int = mul acc x;
    x : int = sub x 1;
  }
  return acc;
};

x : int = fact 5;
_ : unit !output = print x;
