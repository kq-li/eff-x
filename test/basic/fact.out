120
statements executed: 24
